addons/account/models/account_reconcile_model.py

```diff
@@ -317,7 +317,8 @@ def _get_write_off_move_lines_dict(self, st_line, residual_balance):
         lines_vals_list = []
 
         for line in self.line_ids:
-            if not line.account_id or st_line.company_currency_id.is_zero(residual_balance):
+            currency_id = st_line.currency_id or st_line.journal_id.currency_id or self.company_id.currency_id 
```

Comment:
There is no `journal_id` field on `st_line`. Or am I missing something?
```suggestion
            currency_id = st_line.currency_id or self.company_id.currency_id 
```

So maybe this line is incorrect as well:
https://github.com/odoo/odoo/blob/b4d1360ed1d71c8fda44ca0c3b1a80d5a112b93f/addons/account/models/account_reconcile_model.py#L263

Reply:
The `account.bank.statement.line` inherits from `account.move`, this is how it get the fields `journal_id` and `company_currency_id`, I guess...
https://github.com/odoo/odoo/blob/539dfda3e38f8ddb643d87c000391ed058821f6b/addons/account/models/account_bank_statement.py#L480
It's also managed by the `account.bank.statement` model (not in all cases if the account.bank.statement.line not related to a bank statement, don't know if it's possible...) :
https://github.com/odoo/odoo/blob/539dfda3e38f8ddb643d87c000391ed058821f6b/addons/account/models/account_bank_statement.py#L287-L291
Since there are multiple currency_id possible (self.currency_id, self.journal_id.currency_id, self.company_currency_id, self.company_id.currency_id), not sure what's the best to fallback in...

Reply:
indeed, good catch!